### PR TITLE
feat: added Single video page and liked videos page

### PR DESCRIPTION
**Pull request by Chakravarthi0**

### CODE REVIEW

File: src/App.css [hunk 1]
```diff
@@ -45,15 +45,23 @@ a {
...
 }
 
-.videos-container{
+.videos-container {
   display: grid;
-  justify-items:  center;
+  justify-items: center;
   grid-template-columns: repeat(5, 1fr);
   grid-row-gap: 3rem;
   grid-column-gap: 1rem;
```
Comment (msdeshmukh009) [inline src/App.css]: You can use shorthand.

File: src/App.css [hunk 2]
```diff
@@ -45,15 +45,23 @@ a {
...
   grid-template-columns: repeat(auto-fill, minmax(36rem, 1fr));
```
Comment (msdeshmukh009) [inline src/App.css:52]: Grid template columns twice?

File: src/components/video-card/VideoCard.jsx [hunk 3]
```diff
@@ -1,16 +1,24 @@
...
   const [isModalOpen, setIsModalOpen] = useState(false);
 
-  const { _id, title, channel, channelAvatar, views, duration } = video;
+  const { _id, channel, channelAvatar, views, duration, created } = video;
+
+  let { title } = video;
+
+  if (title.length > 34) {
+    title = title.substr(0, 30) + "...";
+  }
```
Comment (msdeshmukh009) [inline src/components/video-card/VideoCard.jsx]:
```css
-webkit-line-clamp: 2;
-webkit-box-orient: vertical;
display: -webkit-box;
overflow: hidden;
```

File: src/context/LikesContext.jsx [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+    loading: false,
+  });
+  const { SET_LIKES, SET_LOADING, SET_ERROR } = likesActions;
+
+  useEffect(() => {
+    (async () => {
+      if (token) {
+        try {
+          likesDispatch({ type: SET_ERROR, payload: false });
+          likesDispatch({ type: SET_LOADING, payload: true });
```
Comment (msdeshmukh009) [inline src/context/LikesContext.jsx:27]: You can do this in a single dispatch as well.

File: src/layouts/aside-navbar/AsideNavBar.jsx [hunk 5]
```diff
@@ -1,39 +1,46 @@
...
-              <span>Watch Later</span>
-            </Link>
-          </li>
-          <li>
-            <Link className="link aside-nav-link" to={"/history"}>
+            <span>Watch Later</span>
+          </Link>
+        </li>
+        <li>
+          <Link className="link aside-nav-link" to={"/history"}>
```
Comment (msdeshmukh009) [inline src/layouts/aside-navbar/AsideNavBar.jsx:36]: You don't need `{` here

File: src/pages/single-video/SingleVideo.jsx [hunk 6]
```diff
@@ -1,43 +1,94 @@
...
+            <div className="single-video-actions">
+              <i
+                className={
+                  `fas pointer fa-thumbs-up ` + (isLiked ? "primary" : "gray")
+                }
+                aria-hidden="true"
+                onClick={
+                  isLiked
+                    ? () => removeFromLikes(video?._id)
+                    : () => addToLikes(video)
```
Comment (msdeshmukh009) [inline src/pages/single-video/SingleVideo.jsx]: You can check user auth.

**Review by msdeshmukh009 — APPROVED** [review]
This is looking good. Please merge it.

**Review by BelideNikhil — APPROVED** [review]
The functionalities work nicely , apart from the suggestions you are good to go!
Clean code and nice CTA

File: src/pages/single-video/SingleVideo.jsx (hunk 6 above)
Comment (Chakravarthi0) [reply src/pages/single-video/SingleVideo.jsx]: done

File: src/App.css (hunk 2 above)
Comment (Chakravarthi0) [reply src/App.css:52]: removed it

File: src/App.css (hunk 1 above)
Comment (Chakravarthi0) [reply src/App.css]: okay

**Pull request merged**